pkg/gadgets/run/tracer/tracer.go

```diff
@@ -376,6 +378,37 @@ func (t *Tracer) attachProgram(gadgetCtx gadgets.GadgetContext, p *ebpf.ProgramS
 		case strings.HasPrefix(p.SectionName, "kretprobe/"):
 			logger.Debugf("Attaching kretprobe %q to %q", p.Name, p.AttachTo)
 			return link.Kretprobe(p.AttachTo, prog, nil)
+		case strings.HasPrefix(p.SectionName, "uprobe/") || strings.HasPrefix(p.SectionName, "uretprobe/"):
+			captureHost := false
+			for _, container := range t.containers {
+				if container.Pid == 1 {
+					captureHost = true
+				}
+			}
+			if !captureHost {
+				return nil, fmt.Errorf("uprobe can only be used with --host at this moment")
+			}
+
+			parts := strings.Split(p.AttachTo, ":")
+			if len(parts) < 2 {
+				return nil, fmt.Errorf("invalid section name %q", p.AttachTo)
+			}
+			if !filepath.IsAbs(parts[0]) {
+				return nil, fmt.Errorf("section name is not an absolute path: %q", parts[0])
+			}
+			executablePath := filepath.Join(host.HostProcFs, "1/root", parts[0])
```

Comment:
I was experimenting with paths such as:
```
SEC("uretprobe//../root/lib64/libc.so.6:malloc")
```
It gets translated to `/proc/1/root/../root/lib64/libc.so.6`.

I am not sure if we should use `filepath.Clean()` to remove the `..` part, or if we should care at all.

I suggest to keep it like that for now. Since we are tracing on the host with full privileges, and we implicitly trust the gadget image, there are no escaping concerns anyway.

For the next PR with support for tracing container workloads, we can think whether we want to prevent escaping, including symlink escaping.

cc @rata

Reply:
SGTM. I'd add a comment so we don't forget about this in the future.

Reply:
I'd mention the escape and that we should use securejoin in the future: https://github.com/cyphar/filepath-securejoin

I have no opinion regarding doing it now vs add a comment and do it later.

Reply:
The next PR will come next week, and it will extend this part by supporting library searching.
I added a TODO in #2563